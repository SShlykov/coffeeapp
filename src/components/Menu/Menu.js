import React from "react";
import {Image, logo, logoBlack} from "../SmallComp";
import {Link} from 'react-router-dom';

const Menu = ({dark}) => {
    let curLogo = logo;
    let clazzName = "header__item";
    let containerClass = "col-lg-6";
    if(dark){
        curLogo = logoBlack;
        clazzName = "footer__item";
        containerClass= "col-lg-6 offset-lg-3"
    }
    return(
            <div className="row">
                <div className={containerClass}>
                    <ul className="header">
                        <Link className={clazzName} to={"/"}>
                                <Image src={curLogo}/>
                        </Link>
                        <Link className={clazzName} to={"/ourcoffee/"}>
                            Our coffee
                        </Link>
                        <Link className={clazzName} to={"/foryourpleasure/"}>
                            For your pleasure
                        </Link>
                        <Link className={clazzName} to={"/contacts/"}>
                            Contact us
                        </Link>
                    </ul>
                </div>
            </div>
    )
};
export default Menu;