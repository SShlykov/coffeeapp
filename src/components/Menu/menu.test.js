import React from 'react';
import Menu from "./Menu";
import {shallow} from 'enzyme';

describe('Testing <Menu/>', () => {
    const menu = shallow(<Menu/>);
    describe('Testing snap & state', () => {
        it('Menu bar have rendered correctly', () => {
            expect(menu).toMatchSnapshot();
        });    
    });
});