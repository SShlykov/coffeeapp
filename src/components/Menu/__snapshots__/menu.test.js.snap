// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Menu/> Testing snap & state Menu bar have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-lg-8"
      >
        <ul
          className="header"
        >
          <Link
            className="header__item"
            replace={false}
            to="/"
          >
            <Image
              src="Logo.svg"
            />
          </Link>
          <Link
            className="header__item"
            replace={false}
            to="/ourcoffee/"
          >
            Our coffee
          </Link>
          <Link
            className="header__item"
            replace={false}
            to="/foryourpleasure/"
          >
            For your pleasure
          </Link>
          <Link
            className="header__item"
            replace={false}
            to="/contacts/"
          >
            Contact us
          </Link>
        </ul>
      </div>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="header"
        >
          <Link
            className="header__item"
            replace={false}
            to="/"
          >
            <Image
              src="Logo.svg"
            />
          </Link>
          <Link
            className="header__item"
            replace={false}
            to="/ourcoffee/"
          >
            Our coffee
          </Link>
          <Link
            className="header__item"
            replace={false}
            to="/foryourpleasure/"
          >
            For your pleasure
          </Link>
          <Link
            className="header__item"
            replace={false}
            to="/contacts/"
          >
            Contact us
          </Link>
        </ul>,
        "className": "col-lg-8",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="header__item"
              replace={false}
              to="/"
            >
              <Image
                src="Logo.svg"
              />
            </Link>,
            <Link
              className="header__item"
              replace={false}
              to="/ourcoffee/"
            >
              Our coffee
            </Link>,
            <Link
              className="header__item"
              replace={false}
              to="/foryourpleasure/"
            >
              For your pleasure
            </Link>,
            <Link
              className="header__item"
              replace={false}
              to="/contacts/"
            >
              Contact us
            </Link>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                src="Logo.svg"
              />,
              "className": "header__item",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "src": "Logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Our coffee",
              "className": "header__item",
              "replace": false,
              "to": "/ourcoffee/",
            },
            "ref": null,
            "rendered": "Our coffee",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "For your pleasure",
              "className": "header__item",
              "replace": false,
              "to": "/foryourpleasure/",
            },
            "ref": null,
            "rendered": "For your pleasure",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Contact us",
              "className": "header__item",
              "replace": false,
              "to": "/contacts/",
            },
            "ref": null,
            "rendered": "Contact us",
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-lg-8"
        >
          <ul
            className="header"
          >
            <Link
              className="header__item"
              replace={false}
              to="/"
            >
              <Image
                src="Logo.svg"
              />
            </Link>
            <Link
              className="header__item"
              replace={false}
              to="/ourcoffee/"
            >
              Our coffee
            </Link>
            <Link
              className="header__item"
              replace={false}
              to="/foryourpleasure/"
            >
              For your pleasure
            </Link>
            <Link
              className="header__item"
              replace={false}
              to="/contacts/"
            >
              Contact us
            </Link>
          </ul>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="header"
          >
            <Link
              className="header__item"
              replace={false}
              to="/"
            >
              <Image
                src="Logo.svg"
              />
            </Link>
            <Link
              className="header__item"
              replace={false}
              to="/ourcoffee/"
            >
              Our coffee
            </Link>
            <Link
              className="header__item"
              replace={false}
              to="/foryourpleasure/"
            >
              For your pleasure
            </Link>
            <Link
              className="header__item"
              replace={false}
              to="/contacts/"
            >
              Contact us
            </Link>
          </ul>,
          "className": "col-lg-8",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="header__item"
                replace={false}
                to="/"
              >
                <Image
                  src="Logo.svg"
                />
              </Link>,
              <Link
                className="header__item"
                replace={false}
                to="/ourcoffee/"
              >
                Our coffee
              </Link>,
              <Link
                className="header__item"
                replace={false}
                to="/foryourpleasure/"
              >
                For your pleasure
              </Link>,
              <Link
                className="header__item"
                replace={false}
                to="/contacts/"
              >
                Contact us
              </Link>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Image
                  src="Logo.svg"
                />,
                "className": "header__item",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "src": "Logo.svg",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Our coffee",
                "className": "header__item",
                "replace": false,
                "to": "/ourcoffee/",
              },
              "ref": null,
              "rendered": "Our coffee",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "For your pleasure",
                "className": "header__item",
                "replace": false,
                "to": "/foryourpleasure/",
              },
              "ref": null,
              "rendered": "For your pleasure",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Contact us",
                "className": "header__item",
                "replace": false,
                "to": "/contacts/",
              },
              "ref": null,
              "rendered": "Contact us",
              "type": [Function],
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
